pvnet/models/base_model.py

```diff
@@ -609,12 +613,48 @@ def _log_forecast_plot(self, batch, y_hat, accum_batch_num, timesteps_to_plot, p
             print(e)
         plt.close(fig)
 
+    def _log_validation_results(self, batch, y_hat, accum_batch_num):
+        """Append validation results to self.validation_epoch_results"""
+
+        y = batch[self._target_key][:, -self.forecast_len :, 0]
+        batch_size = y.shape[0]
+
+        for i in range(batch_size):
+            y_i = y[i].detach().cpu().numpy()
```

Comment:
I think at least some of this detaching and moving to CPU should happen before the loop starts. Doing it in the loop means the CPU and GPU need to communicate with each other a lot more

Reply:
Thanks, good spot, yea ill move this out the loop